usr/src/uts/common/fs/zfs/zap_micro.c

```diff
@@ -201,6 +218,19 @@ zap_name_alloc(zap_t *zap, const char *key, matchtype_t mt)
 	}
 
 	zn->zn_hash = zap_hash(zn);
+
+	if (zap->zap_normflags) {
+		/*
+		 * We *must* use zn_normflags because this normalization is
+		 * what the matching is based on.  (Not the hash!)
+		 */
+		if (zap_normalize(zap, key, zn->zn_normbuf,
```

Comment:
On Mon, Mar 28, 2016 at 6:54 PM, Matthew Ahrens <notifications@github.com>
wrote:

> In usr/src/uts/common/fs/zfs/zap_micro.c
> <https://github.com/openzfs/openzfs/pull/77#discussion_r57647525>:
>
> > @@ -201,6 +218,19 @@ zap_name_alloc(zap_t *zap, const char *key, matchtype_t mt)
> >  	}
> >
> >  	zn->zn_hash = zap_hash(zn);
> > +
> > +	if (zap->zap_normflags) {
> > +		/*
> > +		 * We *must* use zn_normflags because this normalization is
> > +		 * what the matching is based on.  (Not the hash!)
> > +		 */
> > +		if (zap_normalize(zap, key, zn->zn_normbuf,
>
> Don't you need to change zn_key_norm_numints now, since zn_key_norm
> (==zn_normbuf) could be a different length now?
>

Ewww....yeah, good catch.  I believe you are correct.



> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/openzfs/openzfs/pull/77/files/713d9ff5d7adc49d9e5858a05121641625b6f8d3#r57647525>
>